### DISCUSSION TITLE
Spring Boot Microservices as Workflow Steps

### DISCUSSION THREAD

**Discussion by ca2so4u** [argoproj/argo-workflows#9197] [Q&A] [answered]
Hallo Together,

I want to run a Workflow with Spring Boot Microservices as steps. I managed to run an Argo Workflow step as Spring Boot Microservice. But when I do this, that step runs constantly and never finishes. I wonder if its conceptually the right decission, try doing an Argo Workflow with that kind of microservices? Don't I need steps that run and finish to trigger the following steps? Also I wonder how I can manage to have output parameters for the next step if I run with Java?

Could somebody clear my mind up? Thanks in advance!

**Comment by terrytangyuan** [comment, marked as answer]
Check your controller log and see if there's anything wrong there. 

You can use output parameters or artifacts to pass around your steps and consume it in your Java program via some command line flags.
